### PR TITLE
BREAKING CHANGE: Enriches request status

### PR DESCRIPTION
**Pull request by Ntr0**
Adds the request status metadata to the request struct

### CODE REVIEW

File: request.go [hunk 1]
```diff
@@ -42,27 +42,28 @@ type Requests struct {
...
-			Href string `json:"href"`
-		} `json:"requestStatus"`
-	} `json:"metadata"`
-	Properties struct {
-		Method  string      `json:"method"`
-		Headers interface{} `json:"headers"`
-		Body    interface{} `json:"body"`
-		URL     string      `json:"url"`
-	} `json:"properties"`
+	Metadata RequestMetadata `json:"metadata"`
```
Comment (piepmatz) [inline request.go]: run gofmt

File: request.go [hunk 2]
```diff
@@ -42,27 +42,28 @@ type Requests struct {
 	StatusCode int          `json:"statuscode,omitempty"`
 }
 
+type RequestMetadata struct {
+	CreatedDate   time.Time `json:"createdDate"`
+	CreatedBy     string    `json:"createdBy"`
+	Etag          string    `json:"etag"`
+	RequestStatus RequestStatus `json:"requestStatus"`
```
Comment (piepmatz) [inline request.go]: this is actually a breaking change: a request status' type was available as `Type` before, now its `PBType`

File: request.go [hunk 3]
```diff
@@ -42,30 +42,30 @@ type Requests struct {
...
+type RequestMetadata struct {
+	CreatedDate   time.Time     `json:"createdDate"`
+	CreatedBy     string        `json:"createdBy"`
+	Etag          string        `json:"etag"`
+	RequestStatus RequestStatus `json:"requestStatus"`
+}
+
+type RequestProperties struct {
+	Method  string      `json:"method"`
+	Headers interface{} `json:"headers"`
```
Comment (piepmatz) [inline request.go]: now that you changed `Body` to be a string and this is a breaking change anyway, you could probably also make `Headers` `*http.Header` as it is everywhere else

Comment (Ntr0) [reply request.go]: The headers in this struct is a nested dictionary that is part of the properties field. The others are real http.Headers. So it should be left like it is.

Comment (piepmatz) [reply request.go]: ok

**Pull request merged**